### ISSUE TITLE
URI encoded JDBC credentials causes login failures

### ISSUE THREAD

**Issue by fmisir** [spring-cloud/spring-cloud-connectors#232]
Hi,

I upgraded from spring cloud connectors 2.0.1.RELEASE to 2.0.2.RELEASE and got login failures with a oracle database.

After analyzing the issue it turned out that in 2.0.2 the password and username is URI encoded, thus the login fails when the password/username contains special characters like for my case a bracket ']'.
The encoding is done only for Oracle, in the below classes: 

2.0.2.RELEASE: 
https://github.com/spring-cloud/spring-cloud-connectors/blob/2.0.2.RELEASE/spring-cloud-core/src/main/java/org/springframework/cloud/service/common/OracleServiceInfo.java

2.0.1.RELEASE: 

https://github.com/spring-cloud/spring-cloud-connectors/blob/2.0.1.RELEASE/spring-cloud-core/src/main/java/org/springframework/cloud/service/common/OracleServiceInfo.java

Is there any reason why username and password are going to be URI  encoded for building the JDBC 
url?  Is there a workaround?

**Comment by scottfrederick** [comment]
@fmisir There was [an issue](#213) reported against a few databases (SQL Server and Postgres at least) when the username or password had "special" characters. This change was done to resolve that issue. 

> The encoding is done only for Oracle

That's not the case, it's done for all relational DB services - see the [base class](https://github.com/spring-cloud/spring-cloud-connectors/blob/master/spring-cloud-core/src/main/java/org/springframework/cloud/service/common/RelationalServiceInfo.java#L46), [SQL Server override](https://github.com/spring-cloud/spring-cloud-connectors/blob/master/spring-cloud-core/src/main/java/org/springframework/cloud/service/common/SqlServerServiceInfo.java#L22), [DB2 override](https://github.com/spring-cloud/spring-cloud-connectors/blob/master/spring-cloud-core/src/main/java/org/springframework/cloud/service/common/DB2ServiceInfo.java#L22). 

If you've discovered that the Oracle JDBC driver, or the Oracle DB server, doesn't support this encoding then we could back that change out of just the Oracle `buildJdbcUrl` override that you pointed to above. Can you give any more detail on the error that you're getting so we can get more confidence that Oracle seems to behave differently in this regard than SQL Server and Postgres do?

**Comment by fmisir** [comment]
@scottfrederick 

> That's not the case, it's done for all relational DB services - see the base class, SQL Server override, DB2 override.

Okay thank you, i missed that.

Below is the log that I get when i update to 2.0.2.  I'am using Oracle Database 12c Enterprise Edition. 

```
2018-06-21T08:07:19.526+02:00 [OUT] [com.zaxxer.hikari.pool.HikariPool] [HikariPool-1 - Exception during pool initialization.]
2018-06-21T08:07:19.526+02:00 [OUT] java.sql.SQLException: ORA-01017: invalid username/password; logon denied
2018-06-21T08:07:19.526+02:00 [OUT] at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:447)
2018-06-21T08:07:19.526+02:00 [OUT] at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:389)
2018-06-21T08:07:19.526+02:00 [OUT] at oracle.jdbc.driver.T4CTTIoer.processError(T4CTTIoer.java:382)
2018-06-21T08:07:19.526+02:00 [OUT] at oracle.jdbc.driver.T4CTTIfun.processError(T4CTTIfun.java:675)
2018-06-21T08:07:19.526+02:00 [OUT] at oracle.jdbc.driver.T4CTTIoauthenticate.processError(T4CTTIoauthenticate.java:448)
2018-06-21T08:07:19.526+02:00 [APP/PROC/WEB/1] [OUT] at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:513)
```

In may case the special character is a bracket '[' in the username.

**Comment by scottfrederick** [comment]
@fmisir We could test your condition in a couple of ways. 

1) You could remove the username and password encoding from the `OracleServiceInfo` class in a local clone of the project, re-build a snapshot locally, and verify that works in your case. If that works you could submit a PR with that change. 
2) I can make that change to `OracleServiceInfo`, push the change, let it build into a snapshot in repo.spring.io and then you can test with that. 

Let me know if you want me to do the second option.

**Comment by scottfrederick** [comment]
Fixed in b66b97aa7ba597e9a671399075b981162214c6ff
